rapid/src/main/proto/rapid.proto

```diff
@@ -84,9 +72,8 @@ message JoinResponse
 enum JoinStatusCode {
     HOSTNAME_ALREADY_IN_RING = 0;
     UUID_ALREADY_IN_RING = 1;
-    SAFE_TO_JOIN = 2;
-    CONFIG_CHANGED = 3;
-    MEMBERSHIP_REJECTED = 4;
+    SAME_NODE_ALREADY_IN_RING = 2;
```

Comment:
Is there a difference between `HOSTNAME_ALREADY_IN_RING` and `SAME_NODE_ALREADY_IN_RING`?

Reply:
`SAME_NODE_ALREADY_IN_RING` means that the same hostname and UUID pair is already in the ring. I observed this when the join attempt (message) timed out close to the node having been added. In that case, the joiner attempts again with the same UUID at the same node, and in that case I think it is safe to let it effectively join / retrieve the membership list (which is the code in `MembershipService`)

Reply:
Ah, I see. Best to leave a comment clarifying this behavior somewhere.